trino-s3-proxy/src/main/java/io/trino/s3/proxy/server/rest/ParsedS3Request.java

```diff
@@ -13,13 +13,17 @@
  */
 package io.trino.s3.proxy.server.rest;
 
+import io.trino.s3.proxy.server.signing.ParsedAuthorization;
 import jakarta.ws.rs.core.MultivaluedMap;
 
 import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public record ParsedS3Request(
+        ParsedAuthorization parsedAuthorization,
+        Optional<String> session,
```

Comment:
Why do we need `session` in the `ParsedS3Request`? I don't see it used anywhere. Seems wrong to me as it's related to credentials/keys.